sdk/src/Core/Amazon.Runtime/Telemetry/Attributes.cs

```diff
@@ -0,0 +1,84 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using System.Collections.Generic;
+
+namespace Amazon.Runtime.Telemetry
+{
+    /// <summary>
+    /// Represents a collection of key-value pairs for attributes.
+    /// </summary>
+    public class Attributes
+    {
+        private readonly List<KeyValuePair<string, object>> _attributes;
```

Comment:
Why not just use a Dictionary or OrderedDictionary? It would solve some of the problems in this class and give you o(1) access.

Reply:
Used regular Dictionary since the `OrderedDictionary` doesn't match the regular dictionary performance for insertion and deletion, also the order restriction wasn't mentioned in the OTel specifications.